lib/memo_wise.rb

```diff
@@ -200,7 +200,7 @@ def #{method_name}
 
           klass.module_eval <<~HEREDOC, __FILE__, __LINE__ + 1
             def #{method_name}(#{MemoWise::InternalAPI.args_str(method)})
-              _memo_wise_hash = (@_memo_wise[#{index}] ||= {})
+              _memo_wise_hash = (@_memo_wise_#{method_name} ||= {})
```

Comment:
This is wrong. It should be something along the lines of `_memo_wise_hash = #{MemoWise::InternalAPI.memo_wise_hash(klass, method_name)}`, but I can't seem to get any version of that working. Thoughts?